PyInstaller/depend/utils.py

```diff
@@ -354,7 +354,10 @@ def _restorePaths(old):
         else:
             if not include_library(cpath):
                 continue
-            ret.append((cbin, cpath, "BINARY"))
+            if cbin == cpath:
+                ret.append((os.path.basename(cbin), cpath, "BINARY"))
+            else:
+                ret.append((cbin, cpath, "BINARY"))
```

Comment:
Does this change cover the case where:
* 'cbinaries' contain paths with more complex paths (absolute, relative, e.g.  /dir1/dir2/dir3/mylib.so)
* _resolveCtypesImports converts any 'cbin' to absolute path (`cpath`)
* `cbin` is a path relative to the `dist/appname` directory
* without this change some `cbin` might stay absolute and thus this file is not bundled?

What do you think about apply `basename` for all `cbinaries`? Would that work?